knora/excel2xml.py

```diff
@@ -58,7 +57,7 @@ def make_xsd_id_compatible(string: str) -> str:
     return res
 
 
-def find_date_in_string(string: str, calling_resource: str = "") -> Optional[str]:
```

Comment:
There are 2 changes in this method: 
 - param `calling_resource` was a relict from earlier times, which was not used any more.
 - param `string` can actually be None, this case needs to be covered

Reply:
I'm not sure if this is good. If the method does what it says it does, namely "find a date in a string", it should not be possible to hand None for the string...

Reply:
Okay, I changed the signature back to `str`, but I still sanitize the input. That means that the code demands string to be a string, but it is still ready for the case that string is not a string.